pkg/cache/lrucache.go

```diff
@@ -214,13 +214,23 @@ func (c *GitRepoLRUCache) tryEvict() error {
 			break
 		}
 
+		// Check that the LRU element is not part of neverEvictRepos
+		// If it is, find the nearest LRU not part of neverEvictRepos
+		lru := c.dll.Back()
```

Comment:
`lru` as a variable name is abit confusing. `element` may make more sense. Or `lruNode`. Pick your poison (or let me know what you think makes sense).